### PR TITLE
New dependency update tool; updating to new CLI / CoreFx assemblies

### PR DESCRIPTION
**Pull request by shiftylogic**
- Adding support to CoreFxLab for automatic PRs to update CLI and CoreFx assembly versions based on Maestro (dotnet/versions).
- Updated the current version of CLI and assembly versions to LKG.
- Removed explicit setting of RuntimeFrameworkVersion. This gets picked up implicitly by the CLI anyways.

### CODE REVIEW

File: scripts/build.ps1 [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
     [string]$Configuration="Debug",
     [string]$Restore="true",
-    [string]$Channel="preview",
-    [string]$Version="2.0.0-preview1-005675",
+    [string]$Version="<default>",
     [string]$BuildVersion=[System.DateTime]::Now.ToString('eyyMMdd-1')
 )
 
+if ($Version -eq "<default>") {
+    $Version = (Get-Content "$PSScriptRoot\..\DotnetCLIVersion.txt" -Raw).Trim()
```
Comment (ahsonkhan) [inline scripts/build.ps1]: Is use of Get-Content ok? What about using [System.IO.File]::OpenText?

Comment (shiftylogic) [inline scripts/build.ps1]: For this purpose here, this is fine.

Comment (shiftylogic) [inline scripts/build.ps1]: The file has a single line and shouldn't cause any issues with performance or strange newline stuff.

File: scripts/update-dependencies/NuGet.config [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <add key="dotnet-buildtools" value="https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json" />
```
Comment (ahsonkhan) [inline scripts/update-dependencies/NuGet.config]: Why do we need this nuget.config file? Can't we use: https://github.com/dotnet/corefxlab/blob/master/NuGet.Config#L15

File: scripts/update-dependencies.ps1 [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    if ($lastexitcode -ne $null -and $lastexitcode -ne 0) {
+        Write-Error "Failed to install dotnet.exe, exit code [$lastexitcode], aborting build."
+        exit -1
+    }
+}
+
+# Restore the app
+Write-Host "Restoring App $ProjectPath..."
+Invoke-Expression "$DotNetExePath restore `"$ProjectPath`""
+if($LASTEXITCODE -ne 0) { throw "Build failed" }
```
Comment (ahsonkhan) [inline scripts/update-dependencies.ps1]: "Restore failed", instead of Build.

File: scripts/update-dependencies.ps1 [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    Invoke-Expression -Command "$PSScriptRoot\install-dotnet.ps1 -Version $Version -InstallDir $PSScriptRoot\..\dotnet"
+    if ($lastexitcode -ne $null -and $lastexitcode -ne 0) {
+        Write-Error "Failed to install dotnet.exe, exit code [$lastexitcode], aborting build."
+        exit -1
+    }
+}
+
+# Restore the app
+Write-Host "Restoring App $ProjectPath..."
+Invoke-Expression "$DotNetExePath restore `"$ProjectPath`""
```
Comment (ahsonkhan) [inline scripts/update-dependencies.ps1]: Will this restore update-dependencies.csproj succeed without passing a path to Nuget.Config?

Comment (shiftylogic) [inline scripts/update-dependencies.ps1]: It did for me. It is picking up the config at the root.

**Comment by ahsonkhan** [comment]
LGTM

**Comment by davidfowl** [comment]
What does this tool actually do? Does it just update the dependencies.props file? What does it mean by update corefx versions? Is that only for netcoreapp2.0?

**Comment by ahsonkhan** [comment]
Yes, it triggers dotnet-bot to submit a PR to update the versions in the dependencies.props whenever a new CLI or corefx package becomes available.

Similar to other repos: https://github.com/dotnet/corefx/pull/17919

The change in dotnet/versions repo triggers the PRs that update the dependencies:
https://github.com/dotnet/versions/blob/master/build-info/dotnet/cli/master/Latest.txt

@shiftylogic is still working on the last leg of the change.
The final step dotnet-bot does is force push to its own fork and then submit a PR and we are trying to get dotnet-bot to fork corefxlab.

**Comment by davidfowl** [comment]
Please don't update the versions of all corefx packages like the other repos do. It's complete unnecessary. The only things we need to keep up to date are System.Memory and System.Runtime.CompilerServices.Unsafe. Everything else should be using shipping bits unless you need a new API

**Comment by ahsonkhan** [comment]
> The only things we need to keep up to date are System.Memory and System.Runtime.CompilerServices.Unsafe. 

Those are the only ones that will get updated. It won't update the versions of all corefx packages. No worries there :)

**Pull request merged**